src/common/function/src/scalars/math/modulo.rs

```diff
@@ -0,0 +1,241 @@
+            Ok(ConcreteDataType::uint64_datatype())
+        } else {
+            Ok(ConcreteDataType::float64_datatype())
+        }
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::uniform(2, ConcreteDataType::numerics(), Volatility::Immutable)
+    }
+
+    fn eval(&self, _func_ctx: FunctionContext, columns: &[VectorRef]) -> Result<VectorRef> {
+        ensure!(
+            columns.len() == 2,
+            InvalidFuncArgsSnafu {
+                err_msg: format!(
+                    "The length of the args is not correct, expect exactly two, have: {}",
+                    columns.len()
+                ),
+            }
+        );
+        let nums = &columns[0];
+        let divs = &columns[1];
+        let nums_arrow_array = &nums.to_arrow_array();
+        let divs_arrow_array = &divs.to_arrow_array();
+        let array = numeric::rem(nums_arrow_array, divs_arrow_array).context(ArrowComputeSnafu)?;
+
+        let result = match nums.data_type() {
+            ConcreteDataType::Int8(_)
+            | ConcreteDataType::Int16(_)
+            | ConcreteDataType::Int32(_)
+            | ConcreteDataType::Int64(_) => compute::cast(&array, &ArrowDataType::Int64),
+            ConcreteDataType::UInt8(_)
+            | ConcreteDataType::UInt16(_)
+            | ConcreteDataType::UInt32(_)
+            | ConcreteDataType::UInt64(_) => compute::cast(&array, &ArrowDataType::UInt64),
+            ConcreteDataType::Float32(_) | ConcreteDataType::Float64(_) => {
+                compute::cast(&array, &ArrowDataType::Float64)
+            }
+            _ => unreachable!("unexpected datatype: {:?}", nums.data_type()),
+        }
+        .context(ArrowComputeSnafu)?;
+        Helper::try_into_vector(&result).context(error::FromArrowArraySnafu)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use common_error::ext::ErrorExt;
+    use datatypes::value::Value;
+    use datatypes::vectors::{Float64Vector, Int32Vector, StringVector, UInt32Vector};
+
+    use super::*;
+    #[test]
+    fn test_mod_function_signed() {
+        let function = ModuloFunction;
+        assert_eq!("mod", function.name());
+        assert_eq!(
+            ConcreteDataType::int64_datatype(),
+            function
+                .return_type(&[ConcreteDataType::int64_datatype()])
+                .unwrap()
+        );
+        assert_eq!(
+            ConcreteDataType::int64_datatype(),
+            function
+                .return_type(&[ConcreteDataType::int32_datatype()])
+                .unwrap()
+        );
+
+        let nums = vec![18, -17, 5, -6];
+        let divs = vec![4, 8, -5, -5];
+
+        let args: Vec<VectorRef> = vec![
+            Arc::new(Int32Vector::from_vec(nums.clone())),
+            Arc::new(Int32Vector::from_vec(divs.clone())),
+        ];
+        let result = function.eval(FunctionContext::default(), &args).unwrap();
+        assert_eq!(result.len(), 4);
+        for i in 0..3 {
```

Comment:
Is this `0..3` should be `0..4`?

Reply:
Good catch! I haven't noticed this before 👀 

Are you willing to submit a fix?

Reply:
Indeed 🙈!
Thanks @dimbtp!
Feel free to fix it or I can do it in the couple of hours